src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -348,19 +347,6 @@ export function getDiscoverStateContainer({
     // start subscribing to dataStateContainer, triggering data fetching
     const unsubscribeData = dataStateContainer.subscribe();
 
-    // updates saved search when query or filters change, triggers data fetching
-    const filterUnsubscribe = merge(
-      services.data.query.queryString.getUpdates$(),
-      services.filterManager.getFetches$()
-    ).subscribe(async () => {
-      await savedSearchContainer.update({
-        nextDataView: internalStateContainer.getState().dataView,
-        nextState: appStateContainer.getState(),
-        useFilterAndQueryServices: true,
-      });
-      fetchData();
-    });
```

Comment:
further centralized the logic updating state dependent change and potential sources for race conditions covering updating the savedSearch and triggering a fetch in when filter or query changes in `build_state_subscribe.ts`